src/api-v3/paths/payments.ts

```diff
@@ -11,6 +15,42 @@ import { ERRORS } from "../../errors";
 const config = getConfig();
 
 export default function(api: RippleAPI, log: Function): Operations {
+  async function put(req: Request, res: ValidatableResponse, _next: NextFunction): Promise<void> {
+    let signedTransaction = req.body.signedTransaction
+    if (signedTransaction === undefined) {
+      // Do not parse X-address, if present. This endpoint takes transactions fully-formed.
+      // It only signs and submits them. To prepare transactions (converting X-addresses, if necessary),
+      // use GET /v3/preparations/payments.
+
+      if (!api.isValidAddress(req.body.Account)) {
+        throw new Error('Invalid `Account`');
+      }
+
+      if (!config.accounts || !config.accounts[req.body.Account]) {
+        throw ERRORS.ACCOUNT_NOT_CONFIGURED;
+      }
+
+      const accountWithSecret = config.accounts[req.body.Account];
+
+      // Require valid Bearer Token
+      let reqHasValidBearerToken = false;
+      if (!accountWithSecret || 'Bearer ' + accountWithSecret.apiKey != req.headers.authorization) {
+        log(`[401] does not match apiKey, authorization: ${req.headers.authorization}`);
+        throw ERRORS.INVALID_BEARER_TOKEN;
+      } else {
+        reqHasValidBearerToken = true;
+      }
+
+      if (reqHasValidBearerToken) { // In the future, apply velocity limits here
```

Comment:
Why even have the reqHasValidBearerToken` variable?

Reply:
This is to make it more clear that the account secret is not accessed without a valid bearer token. Open to other ways of doing this.

Reply:
Ok, so the argument here is that:
```ts
      let reqHasValidBearerToken = false;
      if (!accountWithSecret || 'Bearer ' + accountWithSecret.apiKey != req.headers.authorization) {
        log(`[401] does not match apiKey, authorization: ${req.headers.authorization}`);
        throw ERRORS.INVALID_BEARER_TOKEN;
      } else {
        reqHasValidBearerToken = true;
      }

      if (reqHasValidBearerToken) { // In the future, apply velocity limits here
                signedTransaction = api.sign(JSON.stringify(req.body), accountWithSecret.secret).signedTransaction;
      }
```

is more clear than:
```ts
      if (!accountWithSecret || 'Bearer ' + accountWithSecret.apiKey != req.headers.authorization {
        log(`[401] does not match apiKey, authorization: ${req.headers.authorization}`);
        throw ERRORS.INVALID_BEARER_TOKEN;
      }


      signedTransaction = api.sign(JSON.stringify(req.body), accountWithSecret.secret).signedTransaction;
```

???

Reply:
Look at the call to `api.sign` and ask "Can this code execute if the request does not have a valid bearer token?"